Projektarbeit III/Assets/Scripts/Player/PlayerController/WallStickingState.cs

```diff
@@ -0,0 +1,83 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class WallStickingState : Interface.IState
+{
+    private Controller controller;
+    private Rigidbody2D rb;
+    private PlayerInput playerInput;
+    private InputAction movementAction;
+    private InputAction jumpAction;
+    private float stickDuration;
+    private float stickTimer;
+    private bool stickingToLeftWall;
+
+    public WallStickingState(Controller controller)
+    {
+        this.controller = controller;
+        rb = controller.GetComponent<Rigidbody2D>();
+        playerInput = controller.GetComponent<PlayerInput>();
+        movementAction = playerInput.actions["Walking"];
+        jumpAction = playerInput.actions["Jumping"];
+        stickDuration = controller.movementEditor.stickDuration;
+        stickTimer = stickDuration;
+        controller.movementEditor.hasJumped = false;
+    }
+
+    public void OnEnter()
+    {
+        Debug.Log("Entered Wall Sticking State");
+
+        // Determine which wall the player is sticking to
+        stickingToLeftWall = controller.IsTouchingLeftWall();
+        rb.linearVelocity = Vector2.zero; // Stop movement initially
+        rb.gravityScale = 0f; // Disable gravity
+        stickTimer = stickDuration; // Initialize the stick timer
+    }
+
+    public void UpdateState()
+    {
+        Vector2 movementInput = movementAction.ReadValue<Vector2>();
+        stickTimer -= Time.deltaTime;
+
+        // Ensure the player presses the correct input for sticking
+        bool correctInput = stickingToLeftWall ? movementInput.x < 0 : movementInput.x > 0;
```

Comment:
This is what the designers wanted but it feels a little weird to control. It should probably include not pressing anything but this is a thing for the review I guess.